File: Anul II/Semestrul II/Medii de proiectare si programare/AgentiiZboruriC/repository/ClientDbRepo.cs
using System.Data.SQLite;
using AgentiiZboruriC.domain;
using log4net.Repository.Hierarchy;

namespace AgentiiZboruriC.repository;

using System;
using System.Collections;
using System.Collections.Generic;
using System.Data.SqlClient;
using System.Linq;
using log4net;

public class ClientDbRepo : Repository<int, Client>
    {
        private readonly DbUtils dbUtils;
        private readonly PersoanaDbRepo persoanaDbRepo;

        public ClientDbRepo(DbUtils dbUtils, PersoanaDbRepo persoanaDbRepo)
        {
            this.dbUtils = dbUtils;
            this.persoanaDbRepo = persoanaDbRepo;
        }

        public Client? FindOne(object o)
        {
            using (SQLiteConnection con = dbUtils.GetConnection())
            using (SQLiteCommand cmd = new SQLiteCommand("SELECT * FROM Client c JOIN Persoana p ON c.id = p.id WHERE c.id=@id", con))
            {
                cmd.Parameters.AddWithValue("@id", (int)o);
                using (SQLiteDataReader result = cmd.ExecuteReader())
                {
                    if (result.Read())
                    {
                        int id = result.GetInt32(0);
                        string adresa = result.GetString(1);
                        string nume = result.GetString(2);
                        string prenume = result.GetString(3);
                        return new Client(id, nume, prenume, adresa);
                    }
                }
            }
            return null;
        }

        public Zbor FindOne(int id)
        {
            throw new NotImplementedException();
        }

        public IEnumerable<Client> FindAll()
        {
            List<Client> clients = new List<Client>();
            using (SQLiteConnection con = dbUtils.GetConnection())
            using (SQLiteCommand cmd = new SQLiteCommand("SELECT * FROM Client c JOIN Persoana p ON c.id = p.id", con))
            {
                using (SQLiteDataReader result = cmd.ExecuteReader())
                {
                    while (result.Read())
                    {
                        int id = result.GetInt32(0);
                        string adresa = result.GetString(1);
                        string nume = result.GetString(2);
                        string prenume = result.GetString(3);
                        clients.Add(new Client(id, nume, prenume, adresa));
                    }
                }
            }
            return clients;
        }

        Optional<Client> Repository<int, Client>.Save(Client entity)
        {
            throw new NotImplementedException();
        }

        public Optional<Client> Delete(int id)
        {
            throw new NotImplementedException();
        }

        Optional<Client> Repository<int, Client>.Update(Client entity)
        {
            throw new NotImplementedException();
        }

        public Client? Save(Client entity)
        {
            using (SQLiteConnection con = dbUtils.GetConnection())
            using (SQLiteCommand cmd = new SQLiteCommand("INSERT INTO Client (id, adresa) VALUES (@id, @adresa)", con))
            {
                cmd.Parameters.AddWithValue("@id", entity.Id);
                cmd.Parameters.AddWithValue("@adresa", entity.Adresa);
                cmd.ExecuteNonQuery();
            }
            return null;
        }

        public Client? Delete(object o)
        {
            return null;
        }

        public Client? Update(Client entity)
        {
            return null;
        }

        public IEnumerable<Client> ChangeEntities(Dictionary<int, Client> entities)
        {
            return null;
        }
    }

File: Anul II/Semestrul II/Medii de proiectare si programare/AgentiiZboruri/Persistence/src/main/java/ro/persistence/ClientDbRepo.java
package ro.persistence;

import ro.model.Client;
import ro.model.Entity;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.*;

public class ClientDbRepo implements Repository{
    private final DbUtils dbUtils;
    private final PersoanaDbRepo persoanaDbRepo;

    public ClientDbRepo(DbUtils dbUtils, PersoanaDbRepo persoanaDbRepo) {
        this.dbUtils = dbUtils;
        this.persoanaDbRepo = persoanaDbRepo;
    }

    @Override
    public Optional findOne(Object o) {
        Connection con = dbUtils.getConnection();
        try (PreparedStatement preStmt = con.prepareStatement("SELECT * FROM Client c JOIN Persoana p ON c.id = p.id WHERE c.id=?")) {
            preStmt.setInt(1, (Integer) o);
            try (ResultSet result = preStmt.executeQuery()) {
                if (result.next()) {
                    Integer id=result.getInt("idP");
                    String adresa=result.getString("adresa");
                    String nume=result.getString("nume");
                    String prenume=result.getString("prenume");
                    Client client = new Client(id, nume, prenume, adresa);
                    return Optional.of(client);
                }
            }
        } catch (SQLException ex) {
            System.out.println("Error DB " + ex);
        }
        return Optional.empty();
    }

    @Override
    public Iterable findAll() {
        Connection con = dbUtils.getConnection();
        List<Client> clients = new ArrayList<>();
        try (PreparedStatement preStmt = con.prepareStatement("SELECT * FROM Client c JOIN Persoana p ON c.id = p.id")) {
            try (ResultSet result = preStmt.executeQuery()) {
                while (result.next()) {
                    Integer id=result.getInt("idP");
                    String adresa=result.getString("adresa");
                    String nume=result.getString("nume");
                    String prenume=result.getString("prenume");
                    Client client = new Client(id, nume, prenume, adresa);
                    clients.add(client);
                }
            }
        } catch (SQLException ex) {
            System.out.println("Error DB " + ex);
        }
        return clients;
    }

    @Override
    public Optional save(Entity entity) {
        Connection con = dbUtils.getConnection();
        try (PreparedStatement preStmt = con.prepareStatement("INSERT INTO Client (id, adresa) VALUES (?, ?)")) {
            Client client = (Client) entity;
            preStmt.setInt(1, client.getId());
            preStmt.setString(2, client.getAdresa());
            preStmt.executeUpdate();
        } catch (SQLException ex) {
            System.out.println("Error DB " + ex);
            return Optional.of(entity);
        }
        return Optional.empty();
    }

    @Override
    public Optional delete(Object o) {
        return Optional.empty();
    }

    @Override
    public Optional update(Entity entity) {
        return null;
    }

    @Override
    public Iterable changeEntities(Map entities) {
        return null;
    }
}